cmd/main.go

```diff
@@ -159,6 +165,7 @@ func main() {
 	}
 	http.Handle("/metrics", promhttp.Handler())
 	if err := httpServer.ListenAndServeTLS(crtFile, keyFile); err != nil {
+		w.Close()
```

Comment:
Would it make sense to `defer w.Close()` directly after the creation, so that it is closed on all code-paths?